src/clf/morpho/morphoDashboardComputer.js

```diff
@@ -0,0 +1,332 @@
+ * 
+ * @param {{liquidationBonusBPS: number, supplyCap: number, LTV: number}} assetParameters Asset parameters including liquidation bonus in basis points, supply cap, and loan-to-value ratio.
+ * @param {string} collateralSymbol The symbol for the collateral asset.
+ * @param {string} baseAsset The base asset symbol.
+ * @param {number} fromBlock The starting block number for the calculation period.
+ * @param {number} endBlock The ending block number for the calculation period.
+ * @param {ethers.providers.StaticJsonRpcProvider} web3Provider The web3 provider for making blockchain calls.
+ * 
+ * @returns {Promise<{
+ *   volatility: number,
+ *   liquidity: number,
+ *   riskLevel: number
+ * }>} An object containing the computed volatility and liquidity of the market, as well as the overall risk level.
+ * 
+ * The function calculates volatility based on historical data up to `endBlock` and computes the average liquidity
+ * within the specified block range (`fromBlock` to `endBlock`). The risk level is derived from these metrics in conjunction
+ * with the provided asset parameters.
+ */
+async function computeMarketRiskLevelBiggestDailyChange(assetParameters, collateralSymbol, baseAsset, fromBlock, endBlock, web3Provider) {
+    const from = collateralSymbol;
+
+
+    // for each platform, compute the volatility and the avg liquidity
+    // only request one data (the biggest span) and recompute the avg for each spans
+    const rollingVolatility = await getRollingVolatility('all', from, baseAsset, web3Provider);
+    const volatilityAtBlock = rollingVolatility.history.filter(_ => _.blockStart <= endBlock && _.blockEnd >= endBlock)[0];
+
+    let volatility = 0;
+    if (volatilityAtBlock) {
+        volatility = volatilityAtBlock.current;
+    } else if (rollingVolatility.latest && rollingVolatility.latest.current) {
+        volatility = rollingVolatility.latest.current;
+    }
+    else {
+        throw new Error('CANNOT FIND VOLATILITY');
+    }
+
+    console.log(`[${from}-${baseAsset}] volatility: ${roundTo(volatility * 100)}%`);
+
+    const toReturn = {
+        volatility,
+        liquidity: 0,
+    };
+
+
+    const oldestBlock = fromBlock;
+    const fullLiquidity = getLiquidityAll(from, baseAsset, oldestBlock, endBlock);
+    const allBlockNumbers = Object.keys(fullLiquidity).map(_ => Number(_));
+
+
+    const blockNumberForSpan = allBlockNumbers.filter(_ => _ >= fromBlock);
+
+    let liquidityToAdd = 0;
+    if (blockNumberForSpan.length > 0) {
+        let sumLiquidityForTargetSlippageBps = 0;
+        for (const blockNumber of blockNumberForSpan) {
+            sumLiquidityForTargetSlippageBps += fullLiquidity[blockNumber].slippageMap[assetParameters.liquidationBonusBPS].quote;
+        }
+
+        liquidityToAdd = sumLiquidityForTargetSlippageBps / blockNumberForSpan.length;
+    }
+
+    toReturn.liquidity += liquidityToAdd;
+    console.log(`[${from}-${baseAsset}] [30d] all dexes liquidity: ${liquidityToAdd}`);
```

Comment:
can you implement this function in the data.interface?

https://github.com/LaTribuWeb3/kinza-dex-history-fetcher/blob/main/src/data.interface/internal/data.interface.liquidity.js#L133-L159

Then you just have to compute the liquidity like that (like in kinza): 

const fullLiquidity = getLiquidityAll(base, quote, blockNumberThirtyDaysAgo, currentBlock);
const averageLiquidityOn30Days = computeAverageSlippageMap(fullLiquidity);